analyses/mutational-signatures/scripts/de_novo_signature_extraction.R

```diff
@@ -0,0 +1,124 @@
+#   --nsignatures_floor 10 \
+#   --nsignatures_ceiling 10 \
+#   --num_iterations 10000 \
+#   --seed 42 \
+#   --output_file results/nsig10_signatures.RDS
+
+#### ---------------------------------------------------------------------------
+
+# Required for command line arguments
+library(optparse)
+
+# Required for signature extraction
+library(deconstructSigs)
+library(sigfit)
+
+# Set up command line options
+option_list <- list(
+  make_option(c("--maf_file"),
+              type = "character",
+              help = "MAF file to be used for mutational signature extraction"),
+  make_option(c("--nsignatures_floor"),
+              type = "integer",
+              default = 10,
+              help = "The lower bound for the number of signatures to be extracted"),
+  make_option(c("--nsignatures_ceiling"),
+              type = "integer",
+              default = 10,
+              help = "The upper bound for the number of signatures to be extracted; should be >= --nsignature_floor"),
+  make_option(c("--num_iterations"),
+              type = "integer",
+              default = 1000,
+              help = "Number of iterations for sampling"),
+  make_option(c("--seed"),
+              type = "integer",
+              default = 42,
+              help = "Seed"),
+  make_option(c("--output_file"),
+              type = "character",
+              help = "Full path to output file (.RDS)"),
+  make_option(c("--plot_output"),
+              type = "character",
+              default = NULL,
+              help = "Full path to goodness-of-fit plot (PDF)")
+)
+
+# Parse command line options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Warn if the ceiling for the number of signatures is less than the floor
+if (opt$nsignatures_ceiling < opt$nsignatures_floor) {
+  warning("Upper bound for signatures is less than lower bound!")
+}
+
+# The nsignatures argument to sigfit::extract_signatures can be a range of
+# values, so we get the vector from the nsignatures_floor and
+# nsignatures_ceiling arguments where if these numbers are the same this will be
+# a vector of length 1
+num_signatures <- opt$nsignatures_floor:opt$nsignatures_ceiling
+
+# Read in the MAF file
+maf <- data.table::fread(opt$maf_file, data.table = FALSE)
+
+# Prep mutations file for signature extraction
+sigs_input <- deconstructSigs::mut.to.sigs.input(
+  mut.ref = maf,
+  sample.id = "Tumor_Sample_Barcode",
+  chr = "Chromosome",
+  pos = "Start_Position",
+  ref = "Reference_Allele",
+  alt = "Allele",
+  bsg = BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38
+)
+
+# Extract signatures
+extracted_signatures <- sigfit::extract_signatures(
+  counts = sigs_input,
+  nsignatures = num_signatures,
+  iter = opt$num_iterations,
+  seed = opt$seed
+)
+
+# If user specifies a plot output, save the goodness-of-fit plot at that 
+# location
+if (!is.null(opt$plot_output)) {
+  pdf(opt$plot_output, width = 7, height = 7)
+  sigfit::plot_gof(extracted_signatures)
```

Comment:
You attached the library but are also using `sigfit::` is that intentional?

Reply:
no, it is not.

Reply:
Yes, for me it was intentional for _clarity_ (e.g., where does the function come from) but not necessary for function.

Reply:
Oh sorry, I had initially gone back and forth with that, so I thought this was still from my part.

Reply:
No worries 😄 I wanted to make sure I answered @cansavvy's question!